docs/how-to/h-upgrade.md

```diff
@@ -104,13 +104,13 @@ kafka/2       active    idle   10.193.41.221        Upgrade completed
 ### Failing upgrade
 
 Before upgrading the unit, the charm will check whether the upgrade can be performed, e.g. this may mean:
-1. Checking that the upgrade from the previous charm revision and Kafka version is allowed
-2. Checking that other external applications that Kafka depends on (e.g. Zookeeper) are running the correct version
+1. Checking that the upgrade from the previous charm revision and Kafka version is allowed.
+2. Checking that other external applications that Kafka depends on (e.g. ZooKeeper) are running the correct version.
 
 Note that these checks are only possible after a refresh of the charm code, and therefore cannot be done upfront (e.g. during the `pre-upgrade-checks` action).
 If some of these checks fail, the upgrade will be aborted. When this happens, the workload may still be operating (as only the operator may have failed) but we recommend to rollback the upgrade as soon as possible. 
 
-In order to roll back the upgrade, re-run steps 2 and 3, using the revision taken in step 1, i.e.
+To roll back the upgrade, re-run steps two and three, using the revision taken in step one, i.e.:
```

Comment:
given we call "Step 1" and "Step X", isn't it clearer to use the numbers? If we still want to use "one", I would update the heading to keep it consistent

Reply:
I like using numbers in text as it seems easier to navigate and highlight important information. But we have to be consistent and follow our style guide - https://docs.ubuntu.com/styleguide/en#numbers. In this particular case, we can justify an exception for the Step numeration but it's a slippery slope.

Reply:
To me this would definitely be included in 

> Exceptions to this rule include numbered lists and units of measurement.

I believe we should be consistent. If we really want to spell out the numbers, we should also do this in the [heading](https://github.com/canonical/kafka-k8s-operator/blob/f27818eccebafcd0188ce3d26c9195dc660e21c6/docs/how-to/h-upgrade.md?plain=1#L24) with "Step one: Collect". 

But really, to me those numbers look like a numbered list, IMHO

Reply:
Let's do numbers for now.
I've added the `#` sign now. It looks good and will help to evade Vale linter warnings.

Reply:
Nice! Sounds good!